lhotse/recipes/gigaspeech.py

```diff
@@ -1,14 +1,37 @@
-from typing import Optional, Sequence, Union
-from lhotse.utils import Pathlike, is_module_available
+from typing import Dict, Union
 
+from lhotse.audio import Recording, RecordingSet
+from lhotse.supervision import SupervisionSegment, SupervisionSet
+from lhotse.utils import is_module_available
 
-def download_gigaspeech(
-        target_dir: Pathlike = '.',
-        dataset_parts: Optional[Union[str, Sequence[str]]] = "XS",
-        force_download: Optional[bool] = False
-) -> None:
-    if is_module_available('speechcolab'):
-        from speechcolab.datasets import gigaspeech
+if not is_module_available('speechcolab'):
+    raise ImportError('To process the GigaSpeech corpus, please install optional dependency: pip install '
+                      'speechcolab')
+else:
+    from speechcolab.datasets.gigaspeech import GigaSpeech
 
-    gigaspeech_data = gigaspeech.GigaSpeech(target_dir)
-    gigaspeech_data.download(dataset_parts, force_download)
+
+def prepare_gigaspeech(
+        gigaspeech: GigaSpeech,
+        subset: str
```

Comment:
how about we add `dataset_parts` like the ones in LibriSpeech with a default of 'auto' or 'all'

https://github.com/lhotse-speech/lhotse/blob/78dc60538e97ec7254ee302313450aed1eae8244/lhotse/recipes/librispeech.py#L69

and return a dict of:

```
{
  "XS": {"recordings": ..., "supervisions": ...},
  "test": {...},
  ...
}
```